File: in_Java/0921 Minimum Add to Make Parentheses Valid.java
class Solution {
    public int minAddToMakeValid(String S) {
        if(S == null || S.length() == 0) return 0;
        Stack<Character> st = new Stack<>();
        for(char c : S.toCharArray()){
            if(c == '(') st.push(c);
            else{
                if(!st.isEmpty() && st.peek() == '(') st.pop();
                else st.push(c);
            }
        }
        return st.size();
    }
}

File: in_Python/0921 Minimum Add to Make Parentheses Valid.py
class Solution(object):
    def minAddToMakeValid(self, S):
        """
        :type S: str
        :rtype: int
        """
        st = []
        for s in S:
            if s == '(':
                st.append(s)
            else:
                if len(st) == 0:
                    st.append(s)
                else:
                    if st[-1] == '(':
                        st.pop()
                    else:
                        st.append(s)
        return len(st)